apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs

```diff
@@ -603,14 +604,16 @@ private object ConvertToClrTypeImpl(Value wireValue, System.Type targetClrType,
                         return null;
                     case Value.KindOneofCase.StringValue:
                         return SpannerNumeric.Parse(wireValue.StringValue);
+                    case Value.KindOneofCase.NumberValue:
+                        return SpannerNumeric.FromDecimal(Convert.ToDecimal(wireValue.NumberValue), LossOfPrecisionHandling.Truncate);
                     default:
                         throw new InvalidOperationException(
                             $"Invalid Type conversion from {wireValue.KindCase} to {targetClrType.FullName}");
                 }
             }
             if (targetClrType == typeof(PgNumeric))
             {
-                if (TypeCode != TypeCode.Numeric || TypeAnnotationCode != TypeAnnotationCode.PgNumeric)
+                if ((TypeCode != TypeCode.Numeric || TypeAnnotationCode != TypeAnnotationCode.PgNumeric) && TypeCode != TypeCode.Float64)
```

Comment:
I think this might be clearer as something like:

```csharp
bool isPgNumeric = TypeCode == TypeCode.Numeric && TypeAnnotationCode == TypeAnnotationCode.PgNumeric;
bool isFloat64 = TypeCode == TypeCode.Float64;

if (!isPgNumeric && !isFloat64)
{
    ...
}
```

Do we even need to check the type code, if the type annotation code is PgNumeric? (Do we ever end up with a PgNumeric annotation for a non-numeric type?)

I think we should also clarify the exception message to indicate that it can only be used for numeric results, and only when working with a Postgres dialect.